src/ui/pokedex-ui-handler.ts

```diff
@@ -889,16 +889,11 @@ export default class PokedexUiHandler extends MessageUiHandler {
       if (this.filterMode && this.filterBar.openDropDown) {
         // CANCEL with a filter menu open > close it
         this.filterBar.toggleDropDown(this.filterBarCursor);
-
-        // if there are possible pokemon go the first one of the list
-        if (numberOfStarters > 0) {
-          this.setFilterMode(false);
-          this.scrollCursor = 0;
-          this.updateScroll();
-          this.setCursor(0);
-        }
         success = true;
-
+      } else if (this.filterMode && !this.filterBar.getFilter(this.filterBarCursor).hasDefaultValues()) {
+        this.filterBar.resetSelection(this.filterBarCursor);
+        this.updateStarters;
```

Comment:
```suggestion
        this.updateStarters();
```
Dunno if this is necessary since your example videos seem to show everything working even though you forgot to call the function.